dags/datasets/agol-datastore.py

```diff
@@ -0,0 +1,303 @@
+],
+"Staffing_by_Command":
+[
+    { "id": "Year", "type": "float8", "info": { "notes": "Fiscal Year" } },
+    { "id": "Type_of_Metric", "type": "text", "info": { "notes": "Type of Metric" } },
+    { "id": "Command", "type": "text", "info": { "notes": "Command" } },
+    { "id": "Category", "type": "text", "info": { "notes": "Category" } },
+    { "id": "Count_", "type": "float8", "info": { "notes": "Budget amount" } },
+],
+}
+common_job_settings = {
+    "description": "Toronto Police Service - budget",
+    "start_date": datetime(2021, 9, 30, 0, 0, 0),
+    "schedule": "@once",
+}
+
+# custom function to create multiple custom dags
+def create_dag(dag_id,
+               dataset,
+               schedule,
+               default_args):
+
+    dag = DAG(dag_id,
+              schedule_interval=schedule,
+              default_args=default_args,
+              tags=['police', 'gary']
+              )
+
+    with dag:
+        # init vars
+        data_filename = "data.json"
+        agol_dataset = dataset["agol_dataset"]
+        package_name=dataset['package_id']
+        resource_name = dataset["name"]
+
+        tmp_dir = CreateLocalDirectoryOperator(
+            task_id = "tmp_dir", 
+            path = TMP_DIR / dag_id
+        )   
+
+        fields_filepath = TMP_DIR / dag_id / "fields.json"
+
+        get_package = GetPackageOperator(
+            task_id="get_package",
+            address=CKAN,
+            apikey=CKAN_APIKEY,
+            package_name_or_id=package_name,
+        )
+
+        def is_resource_new(**kwargs):
+            resource = kwargs["ti"].xcom_pull(task_ids="get_or_create_resource")
+
+            if resource["is_new"]:
+                return "new_resource"
+
+            return "existing_resource"
+
+        new_or_existing = BranchPythonOperator(
+            task_id="new_or_existing", python_callable=is_resource_new,
+        )
+
+        new_resource = DummyOperator(task_id="new_resource", dag=dag)
+        existing_resource = DummyOperator(task_id="existing_resource", dag=dag)
+
+        get_agol_data = AGOLDownloadFileOperator(
+            task_id = "get_agol_data",
+            request_url = base_url + agol_dataset + "/FeatureServer/0/",
+            dir = TMP_DIR / dag_id,
+            filename = data_filename,
+            delete_col = ["geometry", "objectid"]
+            #on_success_callback=task_success_slack_alert,
+        )
+
+        get_or_create_resource = GetOrCreateResourceOperator(
+            task_id="get_or_create_resource",
+            address=CKAN,
+            apikey=CKAN_APIKEY,
+            package_name_or_id=package_name,
+            resource_name=resource_name,
+            resource_attributes=dict(
+                format="csv",
+                package_id=package_name,
+                is_preview=True,
+                url_type="datastore",
+                extract_job=f"Airflow: {package_name}",
+            ),
+        )
+
+        def set_fields(**kwargs):
```

Comment:
I disagree - it would be good and it would save us work as we go on to create many DAGs that will repeat this process. It prevents us from writing redundant code.

That being said, I wont let that stand in the way of us merging this PR - it's something we'll have to consider when we look to configure DAGs with `yaml`